src/vtl.ts

```diff
@@ -481,18 +507,30 @@ export abstract class VTL {
           }
 
           return previousTmp;
-        } else if (
-          isIdentifier(node.expr.expr) &&
-          node.expr.expr.name === "Promise"
-        ) {
+        } else if (isIdentifier(expr.expr) && expr.expr.name === "Promise") {
           throw new SynthError(
             ErrorCodes.Unsupported_Use_of_Promises,
             "Appsync does not support concurrent integration invocation or methods on the `Promise` api."
           );
+        } else if (expr.name.name === "push") {
+          if (
+            node.args.length === 1 &&
+            !isSpreadElementExpr(node.args[0].expr)
+          ) {
+            return `${this.eval(expr.expr)}.add(${this.eval(
+              node.args[0].expr
+            )})`;
+          } else {
+            return `${this.eval(expr.expr)}.addAll(${this.addAll(
+              node.args
+                .map((arg) => arg.expr)
+                .filter((e): e is Expr => e !== undefined)
+            )})`;
+          }
```

Comment:
Seems like this is missing the case when there is a mix of spread and not spread. Wouldn't the logic just be to addAll for spread and add single for other args?

Reply:
Ohh, I see that this.addAll does that magically. Maybe document?